puppeteer-pdf.js

```diff
@@ -1,130 +1,111 @@
-const cli = require("commander");
-const fileUrl = require("file-url");
-const fs = require("fs");
-const isUrl = require("is-url");
-const puppeteer = require("puppeteer");
+// main.js
+import fileUrl from "file-url";
+import isUrl from "is-url";
+import puppeteer from "puppeteer";
 
-cli
-  .version("1.2.0")
-  .option("-p, --path <path>", "The file path to save the PDF to.")
-  .option(
-    "-s, --scale [scale]",
-    "Scale of the webpage rendering.",
-    parseFloat,
-    1
-  )
-  .option("-dhf, --displayHeaderFooter", "Display header and footer.", false)
-  .option(
-    "-ht, --headerTemplate [template]",
-    "HTML template for the print header."
-  )
-  .option(
-    "-ft, --footerTemplate [template]",
-    "HTML template for the print footer."
-  )
-  .option("-pb, --printBackground", "Print background graphics.", false)
-  .option("-l, --landscape", "Paper orientation.", false)
-  .option(
-    "-pr, --pageRanges <range>",
-    "Paper ranges to print, e.g., '1-5, 8, 11-13'. Defaults to the empty string, which means print all pages."
-  )
-  .option(
-    "-f, --format [format]",
-    "Paper format. If set, takes priority over width or height options. Defaults to 'Letter'.",
-    "Letter"
-  )
-  .option(
-    "-w, --width [width]",
-    "Paper width, accepts values labeled with units."
-  )
-  .option(
-    "-h, --heigh [height]",
-    "Paper height, accepts values labeled with units."
-  )
-  .option(
-    "-mt, --marginTop [margin]",
-    "Top margin, accepts values labeled with units."
-  )
-  .option(
-    "-mr, --marginRight [margin]",
-    "Right margin, accepts values labeled with units."
-  )
-  .option(
-    "-mb, --marginBottom [margin]",
-    "Bottom margin, accepts values labeled with units."
-  )
-  .option(
-    "-ml, --marginLeft [margin]",
-    "Left margin, accepts values labeled with units."
-  )
-  .option("-d, --debug", "Output Puppeteer PDF options")
-  .option(
-    "-wu, --waitUntil [choice]",
-    "waitUntil accepts choices load, domcontentloaded, networkidle0, networkidle2. Defaults to 'networkidle2'.",
-    "networkidle2"
-  )
-  .option("-stb, --setTransparentBackground", "Set transparent background.", false)
-  .action(function(required, optional) {
-    // TODO: Implement required arguments validation
-  })
-  .parse(process.argv);
+// src/cli.js
+import { Command } from "commander";
+var program = new Command();
+program.version("1.0.0").option("-p, --path <path>", "The file path to save the PDF to.").option(
+  "-s, --scale [scale]",
+  "Scale of the webpage rendering.",
+  parseFloat,
+  1
+).option("-dhf, --displayHeaderFooter", "Display header and footer.", false).option(
+  "-ht, --headerTemplate [template]",
+  "HTML template for the print header."
+).option(
+  "-ft, --footerTemplate [template]",
+  "HTML template for the print footer."
+).option("-pb, --printBackground", "Print background graphics.", false).option("-l, --landscape", "Paper orientation.", false).option(
```

Comment:
Did you test this with contracts containing photo nodes? 
And attachments (that also renders images)?

Reply:
@ahallora Not yet, I would like to leave it to Paulina XD